sdks/go/pkg/beam/runners/dataflow/translate.go

```diff
@@ -290,18 +290,12 @@ func expandCoGBK(nodes map[int]*outputReference, edge *graph.MultiEdge) ([]*df.S
 	return steps, nil
 }
 
-// TODO(herohde) 5/16/2017: we might need to robustly encode the index into the
-// name, so that we can infer the ordering from the names in the harness.
-
 // translateNodes builds a map from nodeID to the Dataflow representation.
 func translateNodes(edges []*graph.MultiEdge) map[int]*outputReference {
 	nodes := make(map[int]*outputReference)
 	for _, edge := range edges {
 		for i, out := range edge.Output {
-			name := "out"
-			if i > 0 {
-				name = fmt.Sprintf("side%v", i)
-			}
+			name := fmt.Sprintf("side%v", i)
```

Comment:
nit: use "i%v" instead? The main input is not a side input.

Reply:
That's not a nit, this is flat-out wrong. :) I had it right in the other repo.